turbopack/crates/turbo-tasks-backend/src/backend/operation/cleanup_old_edges.rs

```diff
@@ -82,6 +82,11 @@ impl Operation for CleanupOldEdgesOperation {
                                 for &child_id in children.iter() {
                                     task.remove(&CachedDataItemKey::Child { task: child_id });
                                 }
+                                update!(task, ChildrenCount, |count: Option<u32>| {
+                                    // If this underflows, we messed up counting somewhere
+                                    let count = count.unwrap_or_default() - children.len() as u32;
+                                    (count != 0).then_some(count)
```

Comment:
Debug builds will panic on arithmetic overflow, so the subtraction part is fine, but for the `as u32` (which will never panic on overflow):

- If this is part of a hot loop, this should at least have a `debug_assert!` to check the cast.
- If this isn't part of a hot loop, this should use `u32::try_from(children.len()).unwrap()`.

I think the lack of panicking behavior on `as` casts was a language design mistake, [but unfortunately it's too late to reasonably add that to the language](https://internals.rust-lang.org/t/rust-2018-facing-the-cast-problem/8328/3).